File: frontend/src/api/export.ts
import request from './request'

export type ExportMode = 'monthly_summary' | 'category_summary' | 'detail'

export interface ExportRequestPayload {
  mode: ExportMode
  year: number
  month: number
  category?: string
  source_type?: string
}

export interface ExportResult {
  filename: string
  download_url: string
  record_count: number
}

export interface ExportFile {
  filename: string
  size: number
  created_at: string
  mode?: string | null
  download_url: string
}

/**
 * 生成 Excel 导出文件
 */
export function generateExport(data: ExportRequestPayload) {
  return request.post<unknown, ExportResult>('/api/export/generate', data, {
    timeout: 120000,
  })
}

/**
 * 获取导出历史
 */
export function getExportHistory() {
  return request.get<unknown, ExportFile[]>('/api/export/history')
}

/**
 * 删除导出文件
 */
export function deleteExportFile(filename: string) {
  return request.delete<unknown, { success: boolean; filename: string }>(
    `/api/export/files/${encodeURIComponent(filename)}`,
  )
}

/**
 * 下载 URL (后端校验 token，需通过带 Authorization 的 fetch 触发)
 */
export function getExportDownloadUrl(filename: string): string {
  return `/api/export/download/${encodeURIComponent(filename)}`
}

/**
 * 触发浏览器下载（自动附带 token）
 * 1. fetch blob
 * 2. 创建临时 a 标签触发下载
 */
export async function downloadExport(filename: string): Promise<void> {
  const url = getExportDownloadUrl(filename)
  const token = localStorage.getItem('token') || ''
  const res = await fetch(url, {
    headers: {
      Authorization: `Bearer ${token}`,
    },
  })
  if (!res.ok) {
    throw new Error(`下载失败: ${res.status}`)
  }
  const blob = await res.blob()
  const objectUrl = URL.createObjectURL(blob)
  const a = document.createElement('a')
  a.href = objectUrl
  a.download = filename
  document.body.appendChild(a)
  a.click()
  document.body.removeChild(a)
  // 释放 URL
  setTimeout(() => URL.revokeObjectURL(objectUrl), 1000)
}

File: backend/app/schemas/export.py
"""导出相关 Pydantic Schema"""
from __future__ import annotations

from enum import Enum
from typing import Optional

from pydantic import BaseModel, Field


class ExportMode(str, Enum):
    """导出模式"""

    monthly_summary = "monthly_summary"  # 按月度汇总(按日期)
    category_summary = "category_summary"  # 按分类汇总
    detail = "detail"  # 明细表


class ExportRequest(BaseModel):
    """导出请求"""

    mode: ExportMode = Field(ExportMode.detail, description="导出模式")
    year: int = Field(..., ge=2000, le=2999, description="年份")
    month: int = Field(..., ge=1, le=12, description="月份")
    category: Optional[str] = Field(None, description="筛选分类(可选)")
    source_type: Optional[str] = Field(
        None, description="筛选来源类型: pdf / paper (可选)"
    )


class ExportResponse(BaseModel):
    """导出响应"""

    filename: str
    download_url: str
    record_count: int


class ExportFileItem(BaseModel):
    """导出历史文件项"""

    filename: str
    size: int
    created_at: str
    mode: Optional[str] = None
    download_url: str
